connectors/src/connectors/zendesk/lib/id_conversions.ts

```diff
@@ -61,14 +54,53 @@ function _getIdFromInternal(internalId: string, prefix: string): number | null {
     : null;
 }
 
-export function getBrandIdFromInternalId(
+export type InternalIdType =
+  | "brand"
+  | "help-center"
+  | "tickets"
+  | "category"
+  | "article"
+  | "ticket";
+
+export function getIdFromInternalId(
+  connectorId: ModelId,
+  internalId: string
+): { type: InternalIdType | null; objectId: number } {
+  let objectId = getBrandIdFromInternalId(connectorId, internalId);
+  if (objectId) {
+    return { type: "brand", objectId };
+  }
+  objectId = getBrandIdFromHelpCenterId(connectorId, internalId);
+  if (objectId) {
+    return { type: "help-center", objectId };
+  }
+  objectId = getBrandIdFromTicketsId(connectorId, internalId);
+  if (objectId) {
+    return { type: "tickets", objectId };
+  }
+  objectId = getCategoryIdFromInternalId(connectorId, internalId);
+  if (objectId) {
+    return { type: "category", objectId };
+  }
+  objectId = getArticleIdFromInternalId(connectorId, internalId);
+  if (objectId) {
+    return { type: "article", objectId };
+  }
+  objectId = getTicketIdFromInternalId(connectorId, internalId);
+  if (objectId) {
+    return { type: "ticket", objectId };
+  }
+  return { type: null, objectId: -1 };
```

Comment:
Can't you throw an error here if the type is not recognized ? That would avoid checking for null after, and ensure we always  have a `InternalIdType`

Reply:
oh yeah that would be more practical, will do!